applications/GeoMechanicsApplication/tests/cpp_tests/test_process_info_parser.cpp

```diff
@@ -0,0 +1,136 @@
+            },
+            {
+                "process_name": "AuxiliarProcess2",
+                "Parameters":)" + parameterString + R"(
+            }
+        ]
+    }
+    )";
+
+    // When
+    JsonProcessInfoParser parser;
+    const auto actual_result = parser.GetProcessList(process_list);
+
+    // Then
+    const std::vector<ProcessParameters> expected_result = {ProcessParameters{Parameters{parameterString}, "ConstraintProcess1"},
+                                                      ProcessParameters{Parameters{parameterString}, "ConstraintProcess2"},
+                                                      ProcessParameters{Parameters{parameterString}, "LoadProcess1"},
+                                                      ProcessParameters{Parameters{parameterString}, "LoadProcess2"},
+                                                      ProcessParameters{Parameters{parameterString}, "AuxiliarProcess1"},
+                                                      ProcessParameters{Parameters{parameterString}, "AuxiliarProcess2"}};
+
+    KRATOS_EXPECT_EQ(expected_result, actual_result);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GetProcessList_GivesDuplicates_ForProcessesWithIdenticalNames, KratosGeoMechanicsFastSuite)
+{
+    // Given
+    const std::string process_list_with_duplicate_names = R"(
+    {
+      "constraints_process_list":
+      [
+          {
+            "process_name": "ApplyVectorConstraintTableProcess",
+            "Parameters":)" + parameterString + R"(
+          },
+          {
+            "process_name": "ApplyVectorConstraintTableProcess",
+            "Parameters":)" + parameterString + R"(
+          }
+      ]
+    }
+    )";
+
+    // When
+    JsonProcessInfoParser parser;
+    const auto actual_result = parser.GetProcessList(process_list_with_duplicate_names);
+
+    // Then
+    const std::vector<ProcessParameters> expected_result = {ProcessParameters{Parameters{parameterString}, "ApplyVectorConstraintTableProcess"},
+                                                            ProcessParameters{Parameters{parameterString}, "ApplyVectorConstraintTableProcess"}};
+
+    KRATOS_EXPECT_EQ(expected_result, actual_result);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GetProcessList_Throws_WhenProcessDoesNotHaveParameters, KratosGeoMechanicsFastSuite)
+{
+    // Given
+    const std::string process_without_parameters = R"(
+    {
+      "constraints_process_list":
+      [
+          {
+            "process_name": "ApplyVectorConstraintTableProcess"
+          }
+      ]
+    }
+    )";
+
+    // When
+    JsonProcessInfoParser parser;
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(parser.GetProcessList(process_without_parameters), "Getting a value that does not exist. entry string : Parameters")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GetProcessList_ReturnsEmptyString_WhenNoProcessesAreDefined, KratosGeoMechanicsFastSuite)
+{
+    JsonProcessInfoParser parser;
+    KRATOS_EXPECT_EQ(parser.GetProcessList(Parameters{}).size(), 0);
```

Comment:
I would prefer to use the `empty()` member function, i.e.:
```suggestion
    KRATOS_EXPECT_TRUE(parser.GetProcessList(Parameters{}).empty());
```

Reply:
Agreed, that's more readable